server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -691,20 +701,16 @@ public void onTimeout() {
                 throw new AssertionError("update task submitted to MasterService cannot remove master");
             }
         } catch (Exception e) {
-            TimeValue executionTime = TimeValue.timeValueMillis(Math.max(0, TimeValue.nsecToMSec(currentTimeInNanos() - startTimeNS)));
-            if (logger.isTraceEnabled()) {
-                logger.trace(
-                    () -> new ParameterizedMessage(
-                        "failed to execute cluster state update in [{}], state:\nversion [{}], source [{}]\n{}{}{}",
-                        executionTime,
-                        previousClusterState.version(),
-                        taskInputs.summary,
-                        previousClusterState.nodes(),
-                        previousClusterState.routingTable(),
-                        previousClusterState.getRoutingNodes()),
-                    e);
-            }
-            warnAboutSlowTaskIfNeeded(executionTime, taskInputs.summary);
+            logger.trace(
+                new ParameterizedMessage(
+                    "failed to execute cluster state update (on version: [{}], uuid: [{}]) for [{}]\n{}{}{}",
+                    previousClusterState.version(),
+                    previousClusterState.stateUUID(),
+                    taskInputs.summary,
+                    previousClusterState.nodes(),
+                    previousClusterState.routingTable(),
+                    previousClusterState.getRoutingNodes()),
```

Comment:
this call is potentially expensive (as it lazily builds the routing nodes), so I wonder if we should still guard this logging call with `logger.isTraceEnabled()`

Reply:
Ok, I think passing a message supplier is enough to avoid that, see 45c1919.